src/jsonpatcherproxy.js

```diff
@@ -150,176 +90,214 @@ const JSONPatcherProxy = (function() {
           Checking isProxifyingTreeNow ensures this is not happening in the first proxification, 
           but in fact is is a proxified object moved around the tree
           */
-          if (revokableInstance && !instance.isProxifyingTreeNow) {
-            revokableInstance.inherited = true;
-          }
-        }
-
-        // if the new value is an object, make sure to watch it
-        if (
-          newValue &&
-          typeof newValue === 'object' &&
-          newValue._isProxified !== true
-        ) {
-          newValue = instance._proxifyObjectTreeRecursively(
-            newValue,
-            destinationPropKey
-          );
-        }
-        if (typeof newValue === 'undefined') {
-          if (target.hasOwnProperty(key)) {
-            // when array element is set to `undefined`, should generate replace to `null`
-            if (Array.isArray(target)) {
-              //undefined array elements are JSON.stringified to `null`
-              instance.defaultCallback({
-                op: 'replace',
-                path: destinationPropKey,
-                value: null
-              });
-            } else {
-              instance.defaultCallback({
-                op: 'remove',
-                path: destinationPropKey
-              });
-            }
-            return ownReflectSet(instance, target, key, newValue);
-          } else if (!Array.isArray(target)) {
-            return ownReflectSet(instance, target, key, newValue);
-          }
+      if (revokableInstance && !instance.isProxifyingTreeNow) {
+        revokableInstance.inherited = true;
+      }
+    }
+    // if the new value is an object, make sure to watch it
+    if (
+      newValue &&
+      typeof newValue == 'object' &&
+      !instance.proxifiedObjectsMap.has(newValue)
+    ) {
+      instance.parenthoodMap.set(newValue, {
+        parent: target,
+        path: key
+      });
+      newValue = instance._proxifyObjectTreeRecursively(target, newValue, key);
+    }
+    if (typeof newValue == 'undefined') {
+      if (target.hasOwnProperty(key)) {
+        // when array element is set to `undefined`, should generate replace to `null`
+        if (Array.isArray(target)) {
+          //undefined array elements are JSON.stringified to `null`
+          instance.defaultCallback({
+            op: 'replace',
+            path: destinationPropKey,
+            value: null
+          });
+        } else {
+          instance.defaultCallback({
+            op: 'remove',
+            path: destinationPropKey
+          });
         }
-        if (Array.isArray(target) && !Number.isInteger(+key.toString())) {
-          return ownReflectSet(instance, target, key, newValue);
+        const oldValue = instance.proxifiedObjectsMap.get(target[key]);
+        // was tje deleted a proxified object?
+        if(oldValue) { 
+          instance.parenthoodMap.delete(target[key]);
+          instance.proxifiedObjectsMap.delete(oldValue);
```

Comment:
Don't we have to disable traps for proxy?

Reply:
I thought the GC would take of that. But now I think you're right.